packages/@tailwindcss-upgrade/src/utils/packages.ts

```diff
@@ -28,45 +130,56 @@ async function detectPackageManager(base: string): Promise<null | string> {
       let packageJson = JSON.parse(packageJsonContent)
       if (packageJson.packageManager) {
         if (packageJson.packageManager.includes('bun')) {
-          return 'bun'
+          return packageManagers.get(base).get('bun')
         }
         if (packageJson.packageManager.includes('yarn')) {
-          return 'yarn'
+          return packageManagers.get(base).get('yarn')
         }
         if (packageJson.packageManager.includes('pnpm')) {
-          return 'pnpm'
+          return packageManagers.get(base).get('pnpm')
         }
         if (packageJson.packageManager.includes('npm')) {
-          return 'npm'
+          return packageManagers.get(base).get('npm')
         }
       }
     } catch {}
 
     // 2. Check for common lockfiles
     try {
       await fs.access(resolve(base, 'bun.lockb'))
-      return 'bun'
+      return packageManagers.get(base).get('bun')
     } catch {}
     try {
       await fs.access(resolve(base, 'bun.lock'))
-      return 'bun'
+      return packageManagers.get(base).get('bun')
     } catch {}
     try {
       await fs.access(resolve(base, 'pnpm-lock.yaml'))
-      return 'pnpm'
+      return packageManagers.get(base).get('pnpm')
     } catch {}
 
     try {
       await fs.access(resolve(base, 'yarn.lock'))
-      return 'yarn'
+      return packageManagers.get(base).get('yarn')
     } catch {}
 
     try {
       await fs.access(resolve(base, 'package-lock.json'))
-      return 'npm'
+      return packageManagers.get(base).get('npm')
     } catch {}
 
     // 3. If no lockfile is found, we might be in a monorepo
+    let previousBase = base
     base = dirname(base)
+
+    // Already at the root
+    if (previousBase === base) {
```

Comment:
Added this check to prevent an infinite loop because `dirname('/') === '/'`. This is very unlikely that you would run into this but just in case...